protocol/syncer.go

```diff
@@ -219,15 +220,18 @@ const syncerV1 = "/syncer/0.1"
 // enqueueBlock adds the specific block to the peerID queue
 func (s *Syncer) enqueueBlock(peerID peer.ID, b *types.Block) {
 	s.logger.Debug("enqueue block", "peer", peerID, "number", b.Number(), "hash", b.Hash())
-
+	s.peersLock.RLock()
 	p, ok := s.peers[peerID]
+	s.peersLock.RUnlock()
 	if ok {
 		p.appendBlock(b)
 	}
 }
 
 // Broadcast broadcasts a block to all peers
 func (s *Syncer) Broadcast(b *types.Block) {
+	s.peersLock.RLock()
+	defer s.peersLock.RUnlock()
```

Comment:
Please, only lock+unlock around the statements that actually use the resource.
In this case that would be around the for loop on L249.

Although in human time initialization is fast, there is no need to have a resource lock and a thread waiting when the resource is not yet demanded.

Reply:
Irrelevant after commit 9593ada